pandas/core/arrays/integer.py

```diff
@@ -206,7 +206,11 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             mask = mask.copy()
         return values, mask
 
-    values = np.array(values, copy=copy)
+    if hasattr(values, "dtype"):
```

Comment:
hmm, can't you use ``np.asarray`` here to avoid this entirely?

Reply:
Do you mean to replace this whole if-else with `values = np.asarray(values)`?

Reply:
yes

Reply:
can you replace this with np.asarray?

Reply:
No, that doesn't work. I have tried that. 
It will start generating the same problem we started with, It will mangle for arrays having large integers with np.nan values.